Remove Rational newtype, move reducing to Ratio

anttih opened the pull request:
Fixes #28

hdgarrood commented on src/Data/Ratio.purs:
```diff
@@ -1,45 +1,64 @@
-  degree _ = 1
-  div (Ratio a b) (Ratio c d) = Ratio (a * d) (b * c)
+instance euclideanRingRatio :: (Ord a, EuclideanRing a) => EuclideanRing (Ratio a) where
+  degree _ = one
```
Bit pedantic but I would probably use `1` here because it has to be `Int`.

anttih replied:
Yes, good catch.

hdgarrood commented on src/Data/Ratio.purs:
```diff
@@ -1,45 +1,64 @@
 denominator :: forall a. Ratio a -> a
 denominator (Ratio _ b) = b
-
-gcd :: forall a. Eq a => EuclideanRing a => Ratio a -> a
```
Since this has been removed I think you might be able to get rid of the qualified Prelude import?

hdgarrood commented on src/Data/Rational.purs:
```diff
@@ -1,64 +1,17 @@
   ( Rational
-  , runRational
-  , rational
-  , (%)
```
Maybe it would be nice to re-export `%` so that people who have only imported Data.Rational don't need to adjust their imports as much?

anttih replied:
Yes, I was thinking about the same!

anttih commented:
Ok, updated and added a note to the readme.

The great thing about this is that we get support for `Ratio BigInt`s for free, no extra libraries needed.

Since Slamdata is using this library, what do you think about this change @natefaubion?
